client/src/context/ToastContext.tsx

```diff
@@ -0,0 +1,119 @@
+import type { PropsWithChildren } from 'react';
+import { createContext, useContext, useEffect, useState } from 'react';
+import styled, { css, keyframes } from 'styled-components';
+
+type ToastVariant = 'default' | 'success' | 'warning' | 'error';
+
+type ToastContextType = {
+  showToast: (variant: ToastVariant, message: string) => void;
+};
+
+const ToastContext = createContext<ToastContextType | undefined>(undefined);
+
+type ToastProviderPros = PropsWithChildren;
+
+export const ToastProvider = (props: ToastProviderPros) => {
+  const { children } = props;
+  const [toastState, setToastState] = useState<{ variant: ToastVariant; message: string | null }>({
+    variant: 'default',
+    message: null,
+  });
+
+  useEffect(() => {
+    if (toastState.variant && toastState.message !== null) {
+      const timer = setTimeout(() => {
+        setToastState({ variant: 'default', message: null });
+      }, 5000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [toastState]);
+
+  const showToast = (variant: ToastVariant, message: string) => {
+    setToastState({ variant, message });
+  };
+
+  const contextValue = {
+    showToast,
+  };
+
+  return (
+    <ToastContext.Provider value={contextValue}>
+      {children}
+      {toastState.message !== null && (
+        <Container>
+          <ToastMessage $variant={toastState.variant}>{toastState.message}</ToastMessage>
+        </Container>
+      )}
+    </ToastContext.Provider>
+  );
+};
+
+export const useToast = () => {
+  const context = useContext(ToastContext);
+
+  if (!context) {
+    throw new Error('useToast를 올바르게 사용하지 않았습니다.');
```

Comment:
```suggestion
    throw new Error('부모 트리에서 ToastProvider를 사용해주세요');
```
이렇게 메세지를 바꾸는 것이 직관적이지 않을까요!?

Reply:
좋습니닷!